scripts/xpcs_corr_gpu_client.py

```diff
@@ -0,0 +1,109 @@
+    parser.add_argument('--qmap', help='Path to the qmap file',
+                        default='/data/xpcs8/partitionMapLibrary/2019-1/comm201901_qmap_aerogel_Lq0.h5')
+    parser.add_argument('--group', help='Visibility in Search', default=None)
+    parser.add_argument('--deployment','-d', default='hannah-polaris', help=f'Deployment configs. Available: {list(deployment_map.keys())}')
+    parser.add_argument('--corr_gpu_loc', default='/eagle/projects/APSDataAnalysis/XPCS/mchu/xpcs_boost/gpu_corr.py',
+                        help=f'Location of gpu corr processing script')
+
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+    args = arg_parse()
+
+    depl = deployment_map.get(args.deployment)
+    if not depl:
+        raise ValueError(f'Invalid Deployment, deployments available: {list(deployment_map.keys())}')
+
+    depl_input = depl.get_input()
+
+    raw_name = os.path.basename(args.raw)
+    hdf_name = os.path.basename(args.hdf)
+    qmap_name = os.path.basename(args.qmap)
+    dataset_name = hdf_name[:hdf_name.rindex('.')] #remove file extension
+
+    dataset_dir = os.path.join(depl_input['input']['staging_dir'], dataset_name)
+
+    # Generate Destination Pathnames.
+    raw_file = os.path.join(dataset_dir, 'input', raw_name)
+    qmap_file = os.path.join(dataset_dir, 'qmap', qmap_name)
+    #do need to transfer the metadata file because corr will look for it
+    #internally even though it is not specified as an argument
+    input_hdf_file = os.path.join(dataset_dir, 'input', hdf_name)
+    output_hdf_file = os.path.join(dataset_dir, 'output', hdf_name)
+
+    flow_input = {
+        'input': {
+            'pilot': {
+                # This is the directory which will be published
+                'dataset': dataset_dir,
+                'index': '6871e83e-866b-41bc-8430-e3cf83b43bdc',
+                'project': 'xpcs-8id',
+                'source_globus_endpoint': depl_input['input']['globus_endpoint_proc'],
+                # Extra groups can be specified here. The XPCS Admins group will always
+                # be provided automatically.
+                'groups': [args.group] if args.group else [],
+            },
+
+            'transfer_from_clutch_to_theta_items': [
+                {
+                    'source_path': args.raw,
+                    'destination_path': raw_file,
+                },
+                {
+                    'source_path': args.hdf,
+                    'destination_path': input_hdf_file,
+                },
+                {
+                    'source_path': args.qmap,
+                    'destination_path': qmap_file,
+                }
+            ],
+
+            'proc_dir': dataset_dir,
+            'raw_file': raw_file,
+            'qmap_file': qmap_file,
+            'metadata_file': input_hdf_file,
+            'hdf_file': output_hdf_file,
+            'corr_gpu_loc': args.corr_gpu_loc,
```

Comment:
I still don't know why we need to point to an external python script instead of making itself a funcx function.